pkg/backend/httpstate/client/client.go

```diff
@@ -607,16 +607,20 @@ func (pc *Client) PublishPolicyPack(ctx context.Context, orgName string,
 	}
 
 	//
-	// Step 2: Upload the compressed PolicyPack directory to the presigned S3 URL. The PolicyPack is
-	// now published.
+	// Step 2: Upload the compressed PolicyPack directory to the pre-signed object storage service URL.
+	// The PolicyPack is now published.
 	//
 
-	putS3Req, err := http.NewRequest(http.MethodPut, resp.UploadURI, dirArchive)
+	putReq, err := http.NewRequest(http.MethodPut, resp.UploadURI, dirArchive)
 	if err != nil {
 		return "", errors.Wrapf(err, "Failed to upload compressed PolicyPack")
 	}
 
-	_, err = http.DefaultClient.Do(putS3Req)
+	for k := range resp.RequiredHeaders {
```

Comment:
Is it true that this works for `nil` headers? (As would be returned by a service that doesn't yet support or need to populate headers?).

Reply:
That's right https://play.golang.org/p/2U7KzvDu0lj